Source/FieldSolver/WarpXPushFieldsEM.cpp

```diff
@@ -668,9 +668,9 @@ WarpX::ApplyInverseVolumeScalingToCurrentDensity (MultiFab* Jx, MultiFab* Jy, Mu
             for (int imode=1 ; imode < nmodes ; imode++) {
                 // Wrap the current density deposited in the guard cells around
                 // to the cells above the axis.
-                if (rmin == 0. && 1-ishift_z <= i && i <= ngJ-ishift_z) {
-                    Jz_arr(i,j,0,2*imode-1) -= Jz_arr(-ishift_z-i,j,0,2*imode-1);
-                    Jz_arr(i,j,0,2*imode) -= Jz_arr(-ishift_z-i,j,0,2*imode);
+                if (rmin == 0. && 1-ishift_z <= i && i <= ngJ[0]-ishift_z) {
+                    Jz_arr(i,j,0,2*imode-1) -= std::pow(-1, imode+1)*Jz_arr(-ishift_z-i,j,0,2*imode-1);
+                    Jz_arr(i,j,0,2*imode) -= std::pow(-1, imode+1)*Jz_arr(-ishift_z-i,j,0,2*imode);
```

Comment:
Also just double checking: I see that for `Jr_arr` and `Jt_arr` the assignement has a `+= std::pow(-1, imode+1)` (with `+=`), while for `Jz_arr` here it's `-= std::pow(-1, imode+1)` (with `-=`). Is that intended or did the `+=` got mixed with `-=`?

Reply:
Those signs are as intended.